server/api/api.steps.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import { createProxyServer } from 'http-proxy';
+
+// without setting up a second server (with secure and insecure modes), the best way to simulate the proxying of calls is mocking them/verifying the api usage
+type mockProxyServerType = {
+  on: (event: string, handler: expressMiddleware) => void;
+  web: jest.Mock<expressMiddleware>;
+};
+
+const placeholderProxyEvent = jest.fn();
+const successResponseBody = 'OK from mock';
+const mockServerImplementation: (
+  shouldError: boolean
+) => mockProxyServerType = (shouldError) => {
+  const mockProxyEvents = {
+    proxyReq: placeholderProxyEvent,
+    proxyRes: placeholderProxyEvent,
+    error: placeholderProxyEvent,
+  };
+  return {
+    on: jest
+      .fn()
+      .mockImplementation(
+        (event, handler) => (mockProxyEvents[event] = handler)
+      ),
+    web: jest.fn().mockImplementation((req, res) => {
+      mockProxyEvents.proxyReq({}, req, res);
+      if (shouldError) {
+        mockProxyEvents.error({ message: 'mock error' }, req, res);
+      } else {
+        mockProxyEvents.proxyRes(
+          { statusCode: 200, statusMessage: successResponseBody },
+          req,
+          res
+        );
+        res.status(200).send(successResponseBody);
+      }
+    }),
+  };
+}; // unpack the web to trigger on etc
+const createMockServerFn: (shouldError?: boolean) => mockProxyServerType = (
+  shouldError = false
+) => mockServerImplementation(shouldError);
+
+jest.mock('http-proxy', () => ({
+  createProxyServer: jest.fn(),
+}));
+
+import { Then, After, Fusion, And, Before } from 'jest-cucumber-fusion';
+import {
+  stepWhichUpdatesWorld,
+  stepWithWorld,
+} from 'test_common/commonServerSteps';
+import { expressMiddleware } from 'types';
+
+Before(() => {
+  createProxyServer.mockReturnValue(createMockServerFn(false));
+});
+
+After(() => {
+  // reset all mocks to starting state
+  jest.clearAllMocks();
+});
+
+And(
+  'the backend proxy returns an error response',
+  stepWhichUpdatesWorld((world) => {
+    const { context } = world;
+    createProxyServer.mockReturnValue(createMockServerFn(true));
+    return {
+      ...world,
+      context: {
+        ...context,
+        responseWillError: true,
+      },
+    };
+  })
+);
+
+Then(
+  'I get the expected proxied response',
+  stepWithWorld(async (world) => {
+    const { request, context, configuration } = world;
+    const securedConfig = configuration.proxy.transport.cert ? true : false;
+
+    const expectFailure = context.responseWillError ? true : false;
```

Comment:
tertiary that returns boolean?

Reply:
`context.responseWillError` is optional - hence the exists and value check resulting in a boolean value